File: array/reverse.js
// can you create a fucntion tat reverses string

function reverse(str) {
    if (!str || str.length < 3 || typeof str !== "string") {
        return "input is not a string";
    }

    const backwards = [];
    const totalItems = str.length - 1;

    for (let i = totalItems; i >= 0; i--) {
        backwards.push(str[i]);
    };
    return backwards.join('');
};

// const result = reverse("Hi My Name Is Nevin");
// console.log(result);

function reverse1(str) {
    return str.split('').reverse().join("");
};

const result2 = reverse1("Hi My Name is Nevin");
console.log(result2)

File: array/reverse.py
def reverse(input):
    if type(input) != str or len(input) < 2:
        return "input is not a string"
    totalNumber = len(input) - 1
    resultList = []
    for i in range(totalNumber, 0, -1):
        resultList.append(input[i])

    return "".join(resultList)


result = reverse("Hi My Name is Nevin")
# result = reverse(4)
print(result)


def reverse2(input):
    if(type(input) != str or len(input) < 2):
        return "input is not a string"
    return " ".join(input.split(" ")[::-1])


result2 = reverse2("Hi My Name Is Nevin")
print(result2)
